base/operators.jl

```diff
@@ -293,7 +293,7 @@ fldmod1{T<:Real}(x::T, y::T) = (fld1(x,y), mod1(x,y))
 fldmod1{T<:Integer}(x::T, y::T) = (fld1(x,y), mod1(x,y))
 
 # transpose
-transpose(x) = x
+transpose(x) = error("transpose not implemented for $(typeof(x)). Consider permutedims.")
```

Comment:
I think we should keep this PR self-contained for now - you can always open a PR for those others later. But yeah, I think something like `ArgumentError` is more useful than `error()`.